src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -246,8 +246,7 @@ void GcsServer::InitGcsPlacementGroupManager() {
       main_service_, gcs_table_storage_, *gcs_node_manager_,
       /*lease_client_factory=*/
       [this](const rpc::Address &address) {
-        auto node_manager_worker_client = rpc::NodeManagerWorkerClient::make(
-            address.ip_address(), address.port(), client_call_manager_);
+        auto node_manager_worker_client = nm_client_pool_.GetByAddress(address);
```

Comment:
Got it, makes sense. I didn't see `Disconnect()` called for every `core_worker_client_pool` connect call tho, is that expected ? https://github.com/ray-project/ray/pull/9934/files

Reply:
<img width="1664" alt="Screen Shot 2020-11-25 at 11 01 22 AM" src="https://user-images.githubusercontent.com/18510752/100270907-8eb15a00-2f0d-11eb-8d8d-1b513c6ccd53.png">

You can call it whenever the client is removed from the cache. The original behavior is, 

- the client is erased from the cache.
- Since the client is a shared pointer, it is GC'ed.
- And the connection (fd) is closed. 

Not 100% sure if I answered your question. Lmk this answer makes you still confused.